hathor/nanocontracts/runner/runner.py

```diff
@@ -906,6 +910,17 @@ def _validate_context(self, ctx: Context) -> None:
                 if isinstance(action, BaseTokenAction) and action.amount < 0:
                     raise NCInvalidContext('amount must be positive')
 
+    def _validate_reentrancy(self, contract_id: ContractId, method_name: str, method: Any) -> None:
+        """Check whether a reentrancy is happening and whether it is allowed."""
+        assert self._call_info is not None
+        allow_reentrancy = getattr(method, NC_ALLOW_REENTRANCY, False)
+        if allow_reentrancy:
+            return
+
+        for call_record in self._call_info.stack:
```

Comment:
This code runs in O(n) where n is the maximum depth allowed which is 100 right now. We can make it run in O(1) by keeping a `dict[tuple[ContractId, method_name], int]`. Is it worth implementing this optimization?

Reply:
It can get up to 1+2+3+...+100 = 100*101/2 = 5050 comparisons, actually. But I guess it's still not much and shouldn't be a problem. We can always optimize it later. Maybe just leave a note about it?

Reply:
I feel like this can be optimized, but I don't think it should be a blocker. Do we have any idea about the order of magnitude of time it takes when n=100? Being proportional to ~10.000 times a single verification should be *manageable* I think, but certainly not ideal.